document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -682,24 +684,53 @@ public void close() {
   }
 
   static class PostgresResultIteratorWithMetaData extends PostgresResultIterator {
-    protected ResultSetMetaData resultSetMetaData;
 
-    public PostgresResultIteratorWithMetaData(
-        ResultSetMetaData resultSetMetaData, ResultSet resultSet) {
+    public PostgresResultIteratorWithMetaData(ResultSet resultSet) {
       super(resultSet);
-      this.resultSetMetaData = resultSetMetaData;
     }
 
     @Override
     protected Document prepareDocument() throws SQLException, IOException {
+      ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
       int columnCount = resultSetMetaData.getColumnCount();
       Map<String, Object> jsonNode = new HashMap();
       for (int i = 1; i <= columnCount; i++) {
         String columnName = resultSetMetaData.getColumnName(i);
-        jsonNode.put(columnName, MAPPER.readTree(resultSet.getString(i)));
+        String columnValue = resultSet.getString(i);
+        if (StringUtils.isNotEmpty(columnValue)) {
+          JsonNode leafNodeValue = MAPPER.readTree(columnValue);
+          if (isNestedField(columnName)) {
+            handleNestedField(columnName, jsonNode, leafNodeValue);
+          } else {
+            jsonNode.put(columnName, leafNodeValue);
+          }
+        }
       }
       return new JSONDocument(MAPPER.writeValueAsString(jsonNode));
     }
+
+    private boolean isNestedField(String columnName) {
+      return columnName.contains("_dot_") ? true : false;
```

Comment:
1. Any reason we are checking for "_dot_" instead of just "."?
2. Can we define a common static constant for "_dot_"?

Reply:
If we are doing some encoding, can we have a separate Util like `org.hypertrace.core.documentstore.mongo.MongoUtils#encodeKey` for Mongo.

Reply:
> Any reason we are checking for "dot" instead of just "."?

`.` is not valid char in Alias in SQL. So, replacing it with `_dot_`

> Can we define a common static constant for "dot"?

Done